src/inmanta/server/services/resourceservice.py

```diff
@@ -995,3 +996,24 @@ async def versioned_resource_details(
         if not resource:
             raise NotFound("The resource with the given id does not exist")
         return resource
+
+    @handle(methods.discovered_resources_create)
+    async def discovered_resources_create(self, env: data.Environment, discovered_resource_name: str, value: Sequence[dict]):
+        discovered_resource = data.DiscoveredResources(
+            environment=env, discovered_resource_name=discovered_resource_name, values=value
+        )
+        (column_names, values) = discovered_resource._get_column_names_and_values()
+        column_names_as_sql_string = ",".join(column_names)
+        values_as_parameterize_sql_string = "$1,$2,$3"
+        query = f"""INSERT INTO {discovered_resource.table_name()} ({column_names_as_sql_string}) VALUES ({values_as_parameterize_sql_string})
+                ON CONFLICT (environment,discovered_resource_name) DO UPDATE SET values = $3;
```

Comment:
We need to be careful here not to overwrite too greedily (as implemented here I think). Related to my comment about the primary key.